### PR TITLE
fix(cmd/influxd): prevent panic on upgrade with V1 users and no V1 conf

### LINKED ISSUE #20545: influxd upgrade panics when V1 users are present and no V1 config is given
__Steps to reproduce:__

1. Create some users in a V1 `influxd` instance
2. Use V2 `influxd upgrade` to upgrade the V1 instance, _without_ passing the `--config-file` option

__Expected behavior:__
Upgrade succeeds.

__Actual behavior:__
Upgrade fails with a panic:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x41 pc=0x20cb197]

goroutine 1 [running]:
github.com/influxdata/influxdb/v2/cmd/influxd/upgrade.runUpgradeE(0xc0000f1600, 0xc000166000, 0xc00090fd00, 0x0, 0x0)
        /home/circleci/go/src/github.com/influxdata/influxdb/cmd/influxd/upgrade/upgrade.go:481 +0x1657
github.com/influxdata/influxdb/v2/cmd/influxd/upgrade.NewCommand.func1(0xc0000f1600, 0xc0001b2a80, 0x0, 0x17, 0x0, 0x0)
        /home/circleci/go/src/github.com/influxdata/influxdb/cmd/influxd/upgrade/upgrade.go:151 +0x3f
github.com/spf13/cobra.(*Command).execute(0xc0000f1600, 0xc0001b2900, 0x17, 0x18, 0xc0000f1600, 0xc0001b2900)
        /home/circleci/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:842 +0x47c
github.com/spf13/cobra.(*Command).ExecuteC(0xc0000f0dc0, 0xc00090ff48, 0x1, 0x1)
        /home/circleci/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        /home/circleci/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
main.main()
        /home/circleci/go/src/github.com/influxdata/influxdb/cmd/influxd/main.go:41 +0x31c
```

### PR DESCRIPTION
**Pull request by danxmoran**
Closes #20545 

I reworked our existing e2e test for `influxd upgrade` to actually cover the main body of the command.

### CODE REVIEW

File: cmd/influxd/upgrade/upgrade.go [hunk 1]
```diff
@@ -332,18 +335,15 @@ func runUpgradeE(cmd *cobra.Command, options *options, verbose bool) error {
...
-		return err
+		return nil, err
 	}
 	if verbose {
 		log.Warn("--verbose is deprecated, use --log-level=debug instead")
 	}
+	return log, nil
+}
 
-	// This command is executed multiple times by test code. Initialization can happen only once.
```
Comment (danxmoran) [inline cmd/influxd/upgrade/upgrade.go:341]: The upgrade process doesn't use flux, I think this was bad copy-pasta

File: cmd/influxd/upgrade/upgrade.go [hunk 2]
```diff
@@ -361,8 +361,9 @@ func runUpgradeE(cmd *cobra.Command, options *options, verbose bool) error {
 		}
 	}
 
-	var v1Config *configV1
+	v1Config := &configV1{}
```
Comment (danxmoran) [inline cmd/influxd/upgrade/upgrade.go:364]: This (making sure `v1Config` isn't `nil`) is the fix for the panic

Comment (danxmoran) [reply cmd/influxd/upgrade/upgrade.go:364]: The panic was occurring on [this line](https://github.com/influxdata/influxdb/pull/20548/files#diff-e5c52e2be1a2771a344e5d0402468658d34a70fe9ba50453edb3b3b3e97a4eaeR476) (`nil` dereference on `!v1Config.Http.AuthEnabled`)

**Review by lesam — APPROVED** [review]
lgtm

**Pull request merged**